experimenter/experimenter/experiments/models.py

```diff
@@ -1137,6 +1137,13 @@ def get_firefox_min_version_display(self):
     def get_firefox_max_version_display(self):
         return self.firefox_max_version.replace("!", "0")
 
+    @property
+    def kinto_collection(self):
+        # Note: this can throw if there are conflicting features targeting
+        # different collections.
+        if self.application_config:
```

Comment:
What are the conditions we'd throw here?  Shouldn't we always promote to secure collection if any feature needs it and just leave it at that?

Reply:
the review serializer will flag recipes if they target multiple collections (ie contain secure and non-secure features)

Reply:
see https://github.com/mozilla/experimenter/blob/898b5520eb1c6e8c09b7395e22e3a71e578c4265/experimenter/experimenter/experiments/api/v5/serializers.py#L1682

https://github.com/mozilla/experimenter/blob/898b5520eb1c6e8c09b7395e22e3a71e578c4265/experimenter/experimenter/experiments/api/v5/serializers.py#L1715

desktop also wont accept a recipe from the secure collection that contains non-secure features and vice versa.

Reply:
Oh yes okay, I spaced that it was a design constraint to not mix secure and non secure features.  Then yes that makes sense 👍